heartbeat/storage-mon.in

```diff
@@ -0,0 +1,259 @@
+: ${OCF_RESKEY_CRM_meta_interval:=0}
+: ${OCF_RESKEY_drives:="/dev/sda"}
+: ${OCF_RESKEY_io_timeout:="10"}
+: ${OCF_RESKEY_inject_errors:=""}
+: ${OCF_RESKEY_state:="${HA_RSCTMP%%/}/storage-mon-${OCF_RESOURCE_INSTANCE}.state"}
+
+#######################################################################
+
+meta_data() {
+	cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="storage-mon">
+<version>1.0</version>
+
+<longdesc lang="en">
+System health agent that checks the storage I/O status of the given drives and
+updates the #health-storage attribute. Usage is highly recommended in combination
+with storage-mon monitoring agent. The agent currently support a maximum of 25
+devices per instance.
+</longdesc>
+<shortdesc lang="en">storage I/O health status</shortdesc>
+
+<parameters>
+
+<parameter name="state" unique="1">
+<longdesc lang="en">
+Location to store the resource state in.
+</longdesc>
+<shortdesc lang="en">State file</shortdesc>
+<content type="string" default="${OCF_RESKEY_state}" />
+</parameter>
+
+<parameter name="drives" unique="0">
+<longdesc lang="en">
+The drive(s) to check as a SPACE separated list. Enter the full path to the device, e.g. "/dev/sda".
+</longdesc>
+<shortdesc lang="en">Drives to check</shortdesc>
+<content type="string" default="${OCF_RESKEY_drives}" />
+</parameter>
+
+<parameter name="io_timeout" unique="1">
+<longdesc lang="en">
+Specify disk I/O timeout in seconds. Minimum 1, recommeded 10 (default).
+</longdesc>
+<shortdesc lang="en">Disk I/O timeout</shortdesc>
+<content type="integer" default="${OCF_RESKEY_io_timeout}" />
+</parameter>
+
+<parameter name="inject_errors" unique="1">
+<longdesc lang="en">
+Used only for testing! Specify % of I/O errors to simulate drives failures.
+</longdesc>
+<shortdesc lang="en">Specify % of I/O errors to simulate drives failures</shortdesc>
+<content type="integer" default="${OCF_RESKEY_inject_errors}" />
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start"        timeout="10s" />
+<action name="stop"         timeout="120s" />
+<action name="monitor"      timeout="120s" interval="30s" start-delay="0s" />
+<action name="meta-data"    timeout="5s" />
+<action name="validate-all" timeout="10s" />
+</actions>
+</resource-agent>
+END
+	return $OCF_SUCCESS
+}
+
+#######################################################################
+
+storage-mon_usage() {
+	cat <<END
+usage: $0 {start|stop|monitor|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+	return $1
+}
+
+storage-mon_init() {
+	#Test for presence of storage_mon helper
+	if [ ! -x "$STORAGEMON" ] ; then
+		ocf_log err "${STORAGEMON} not installed."
+		exit $OCF_ERR_INSTALLED
+	fi
+
+	i=0
+	for DRIVE in ${OCF_RESKEY_drives}; do
+		if [ ! -e "$DRIVE" ] ; then
+			ocf_log err "${DRIVE} not found on the system"
+			exit $OCF_ERR_INSTALLED
```

Comment:
I think _CONFIGURED is better for this and the rest of the returns in this function..